Update timeseries scale to align with the data's week boundaries

paulrosenzweig opened the pull request:
Fixes #13516

Like formatted ranges, ticks also rely on finding the boundary points between weeks. We had already updated the formatting code to check for an offset between the data's week boundary and the clients. I took the same approach here and made our timeseries scale aware of how many days to shift the ticks.

paulrosenzweig committed:
update timeseries scale to align with the data's week boundaries

flamber commented:
This only fixes it in English - testing with Spanish/German still shows the one-off - guess it's more related to #11319 than I expected
![image](https://user-images.githubusercontent.com/1447303/96499207-69b23300-124d-11eb-8fce-350d4c55c14a.png)

nemanjaglumac committed:
Add repro for #13516

Notes:
- test should fail until the issue is resolved,
- please unskip the test with the merge.

nemanjaglumac committed:
Fix linter issue

paulrosenzweig committed:
use moment-timezone in key places

paulrosenzweig commented:
Good catch @flamber! I think it fixes #11319 now too.

paulrosenzweig committed:
import correct package in test

flamber reviewed (approved):
Si señor - arregla los dos problemas :tada:
I'm not qualified to completely sign off on this, but fixes what is says on the can and haven't found any negative side-effects.
![image](https://user-images.githubusercontent.com/1447303/96506466-f2ce6780-1257-11eb-94f4-2e5885f13722.png)

paulrosenzweig committed:
switch another test to match moment module
